wynnset/tapestry-wp
# Allow tapestry import/export/duplicate

Closes #506 

* Updates the `POST /tapestries` endpoint to allow sending a JSON representation of a Tapestry as the request body
  * This endpoint will create a new Tapestry and copy over the nodes and links to the new Tapestry, assigning them new IDs in the process
  * Returns the newly created Tapestry
* Add an export button to the settings modal that converts the current tapestry to a json file and automatically downloads it

Most of these changes were made so that the Cypress tests may be refactored to be more modular and run more independently. As a result, things like styling and adding an import button is delegated to a future PR.

**Testing**

* Open a Tapestry
* Open the settings modal, then go to the "Advanced" tab
* Click "Export Tapestry"
* A download prompt should open with the filename being `title-of-tapestry.json`
* Open the resulting JSON file and verify the contents look correct
